test/rules-google-enhanced-ecommerce-fullstory.spec.ts

```diff
@@ -335,6 +335,17 @@ describe('Ruleset: Google Analytics Enhanced Ecommerce to FullStory', () => {
         expectEqual(eventName, 'refund');
         expectEqual(payload.id, 'T12345');
       });
+
+      it('gracefully handles a "clear" event', async () => {
+        await testHarness.execute(() => {
+          globalThis.dataLayer.push({
+            ecommerce: null,
+          });
+        });
+
+        const event = await testHarness.popEvent();
+        expectEqual(event, undefined);
```

Comment:
Nit: Elsewhere when looking for no events to be triggered, we're:
- Setting a timeout of 500ms via `await testHarness.popEvent(500);`. In local testing usually 250ms is enough time to trigger any events so we've been doubling that to verify that no events were triggered.
- Using `expect(event).to.be.undefined;`; do you have a preference for which form to standardize on?

Reply:
- The `500` ms timeout makes sense.  The `DataHandler.DefaultDebounceTime` is `250` ms so I'd expect that to be the floor of time needed to wait.  The need to wait applies only to shimmed property observation, which is why I also added the unit tests to `selector` and`QueryOperator` since those are synchronous.
- We've used `.to.be.undefined` in the past.  I went with `expectEqual` because that helper function was added to remove ambiguity from doing shallow versus deep comparison.  There are 4 equality tests in mocha, and in most comparisons of DLO, we want deep comparisons since we should be making copies of objects from the data layer.  So I'd probably say `expectEqual` since we'd want to use it as a best practice.